tests/testsuite/lto.rs

```diff
@@ -517,45 +579,48 @@ fn cdylib_and_rlib() {
     );
     verify_lto(&output, "foo", "--crate-type bin", Lto::Run(None));
     p.cargo("test --release -v")
-        .with_stderr_unordered(
-            "\
+        .with_stderr_data(
+            str![[r#"
 [FRESH] registry v0.0.1
 [FRESH] registry-shared v0.0.1
-[FRESH] bar v0.0.0 [..]
-[COMPILING] foo [..]
-[RUNNING] `rustc --crate-name foo [..]-C lto [..]--test[..]
-[RUNNING] `rustc --crate-name a [..]-C lto [..]--test[..]
-[FINISHED] [..]
-[RUNNING] [..]
-[RUNNING] [..]
-",
+[FRESH] bar v0.0.0 ([ROOT]/foo/bar)
+[COMPILING] foo v0.0.0 ([ROOT]/foo)
+[RUNNING] `rustc --crate-name foo [..] -C lto --test [..]`
+[RUNNING] `rustc --crate-name a [..] -C lto --test [..]`
```

Comment:
Could we put `[..]` back between `-C lto` and `--test`? We don't want snapshots to depend on the order of these flags.

Also, the missing space around `[..]` is important for not overly-matching the order

```suggestion
[RUNNING] `rustc --crate-name foo [..]-C lto [..]--test [..]`
[RUNNING] `rustc --crate-name a [..]-C lto [..]--test [..]`
```

(ditto elsewhere throughout this PR)